components/org.wso2.micro.integrator.initializer/src/main/java/org/wso2/micro/integrator/initializer/handler/transaction/TransactionHandlerComponent.java

```
@@ -0,0 +1,189 @@
+    private static final int DEFAULT_SCHEDULED_PERIOD_INTERVAL = 60;
+    private static final String SCHEDULED_PERIOD = "transactionCountScheduledPeriod";
+
+    private static final String TRANSACTION_DATA_SOURCE = "transactionDataSource";
+    private static final String TRANSACTION_CONFIG_SECTION = "transaction";
+    private static final String TRANSACTION_CONFIG_NAME = "data_source";
+
+    private ScheduledExecutorService trnCountWriterTaskScheduler;
+
+    public void start(DataSourceService dataSourceService) throws TransactionException, DataSourceException {
+        String nodeId = generateRandomId();
+
+        // initialize cipher for encryption needs.
+        Cipher cipher = CryptoUtil.initializeCipher();
+
+        DataSource dataSource = getTransactionDataSource(dataSourceService);
+
+        // initialize Transaction store.
+        TransactionStore transactionStore = new TransactionStore(dataSource, nodeId, cipher);
+
+        // start and schedule transaction writer task.
+        scheduleTransactionWriterTask(transactionStore);
+    }
+
+    /**
+     * Get the time rate for running the transaction writer scheduled task.
+     *
+     * @return - scheduled period.
+     */
+    private int getTransactionCountScheduledPeriod() {
+        String scheduledPeriodStr = System.getProperty(SCHEDULED_PERIOD);
+        if (StringUtils.isEmpty(scheduledPeriodStr)) {
+            scheduledPeriodStr = System.getenv(SCHEDULED_PERIOD);
+            if (StringUtils.isEmpty(scheduledPeriodStr)) {
+                Object scheduledPeriodObject = ConfigParser.getParsedConfigs().get("" + "." + "");
+                if (null != scheduledPeriodObject) {
+                    scheduledPeriodStr = scheduledPeriodObject.toString();
+                }
+            }
+        }
+        try {
+            return Integer.parseInt(scheduledPeriodStr);
+        } catch (NumberFormatException e) {
+            return DEFAULT_SCHEDULED_PERIOD_INTERVAL;
```

Should we issue a WARN here? User input is wrong. We should not silently failover to default value.

---

+1. Added a warn log.